src/test/proxy/worker_proxy.py

```diff
@@ -0,0 +1,253 @@
+        self._job_1.status = JobStatus.QUEUED
+        self._uid_2 = "job-2"
+        self._job_2 = Job(
+            owner_id=1003,
+            email=None,
+            scheduling_constraints=JobSchedulingConstraints(
+                priority=JobPriority.MEDIUM, is_preemtible=True, special_resources=[]
+            ),
+            docker_context=DockerContext(dockerfile_source="", mount_points=[]),
+            docker_constraints=DockerConstraints(cpu_threads=1, memory=1024)
+        )
+        self._job_2.uid = self._uid_2
+        self._job_2.status = JobStatus.QUEUED
+        self._uid_unknown = "DEADBEEF"
+        self._empty_worker: IWorkerProxy = None
+        self._busy_worker: IWorkerProxy = None
+
+    @skipIfAbstract
+    def test_dispatch_to_empty_worker(self) -> None:
+        response_1 = self._empty_worker.dispatch_job(self._job_1)
+        self.assertTrue(response_1.is_success)
+        response_2 = self._empty_worker.dispatch_job(self._job_2)
+        self.assertTrue(response_2.is_success)
+
+    @skipIfAbstract
+    def dispatch_to_busy_worker(self) -> None:
+        response_1 = self._busy_worker.dispatch_job(self._job_1)
+        self.assertFalse(response_1.is_success)
+        response_2 = self._busy_worker.dispatch_job(self._job_2)
+        self.assertFalse(response_2.is_success)
+
+    @skipIfAbstract
+    def test_dispatch_same_job_twice_1(self) -> None:
+        response_1 = self._empty_worker.dispatch_job(self._job_1)
+        self.assertTrue(response_1.is_success)
+        response_2 = self._empty_worker.dispatch_job(self._job_1)
+        self.assertFalse(response_2.is_success)
+
+    @skipIfAbstract
+    def test_dispatch_same_job_twice_2(self) -> None:
+        response_1 = self._empty_worker.dispatch_job(self._job_2)
+        self.assertTrue(response_1.is_success)
+        response_2 = self._empty_worker.dispatch_job(self._job_2)
+        self.assertFalse(response_2.is_success)
+
+    @skipIfAbstract
+    def test_cancel_empty_worker(self) -> None:
+        response_1 = self._empty_worker.cancel_job(self._uid_1)
+        self.assertFalse(response_1.is_success)
+        response_2 = self._empty_worker.cancel_job(self._uid_2)
+        self.assertFalse(response_2.is_success)
+
+    @skipIfAbstract
+    def test_cancel_busy_worker(self) -> None:
+        response_1 = self._busy_worker.cancel_job(self._uid_1)
+        self.assertTrue(response_1.is_success)
+        response_2 = self._busy_worker.cancel_job(self._uid_2)
+        self.assertTrue(response_2.is_success)
+
+    @skipIfAbstract
+    def test_cancel_same_job_twice_1(self) -> None:
+        response_1 = self._busy_worker.cancel_job(self._uid_1)
+        self.assertTrue(response_1.is_success)
+        response_2 = self._busy_worker.cancel_job(self._uid_1)
+        self.assertFalse(response_2.is_success)
+
+    @skipIfAbstract
+    def test_cancel_same_job_twice_2(self) -> None:
+        response_1 = self._busy_worker.cancel_job(self._uid_2)
```

Comment:
same as above, duplicate tests.

Reply:
Also the same things below, I know these are tests, but shouldn't we make them more concise?

Reply:
For example, you could iterate over all jobs, and try to pause them on both machines, and expect success depending on whether the job is in proxy._jobs (maybe make that public)

Reply:
I think unit tests should be kept as simple as possible.
You could come up with some approach that took less lines of code but I think that would also be more difficult to debug if any of the tests fail.
If you just have a few lines of code with no if statements and no loops I think that is ideal for a test.

Reply:
I was just saying what I think is standard practice - remove repetitive code as much as possible: https://stackoverflow.com/questions/129693/is-duplicated-code-more-tolerable-in-unit-tests

I would not hold the PR from being merged though, especially not over this.